src/insertItem.ts

```diff
@@ -0,0 +1,449 @@
+        let index = templatePart.span.endIndex;
+        let text = JSON.stringify(data, null, '\t');
+        let indentedText = this.indent(`\r\n${text}\r\n`, 2);
+        await this.insertText(textEditor, index, `${indentedText}\t`, true);
+    }
+
+    private async getFunctionNamespace(): Promise<FunctionNameSpace> {
+        let namespaceName = await this.ui.showInputBox({ prompt: "Name of namespace?" });
+        let namespace = {
+            namespace: namespaceName,
+            members: await this.getFunctionMembers()
+        };
+        return namespace;
+    }
+
+    // tslint:disable-next-line:no-any
+    private async getFunctionMembers(): Promise<any> {
+        let functionName = await this.ui.showInputBox({ prompt: "Name of function?" });
+        let functionDef = await this.getFunction();
+        // tslint:disable-next-line:no-any
+        let members: any = {};
+        // tslint:disable-next-line:no-unsafe-any
+        members[functionName] = functionDef;
+        return members;
+    }
+
+    private async insertText(textEditor: vscode.TextEditor, index: number, text: string, setCursor: boolean = false): Promise<number> {
+        text = this.formatText(text, textEditor);
+        let pos = textEditor.document.positionAt(index);
+        await textEditor.edit(builder => builder.insert(pos, text));
+        textEditor.revealRange(new vscode.Range(pos, pos), vscode.TextEditorRevealType.Default);
+        if (text.lastIndexOf(insertCursorText) >= 0) {
+            let insertedText = textEditor.document.getText(new vscode.Range(pos, textEditor.document.positionAt(index + text.length)));
+            let cursorPos = insertedText.lastIndexOf(insertCursorText);
+            let newIndex = index + cursorPos + insertCursorText.length / 2;
+            let pos2 = textEditor.document.positionAt(newIndex);
+            textEditor.selection = new vscode.Selection(pos2, pos2);
+            return newIndex;
+        }
+        return 0;
+    }
+
+    private formatText(text: string, textEditor: vscode.TextEditor): string {
+        if (textEditor.options.insertSpaces === true) {
+            text = text.replace(/\t/g, ' '.repeat(Number(textEditor.options.tabSize)));
+        } else {
+            text = text.replace(/ {4}/g, '\t');
+        }
+        if (textEditor.document.eol === vscode.EndOfLine.LF) {
+            text = text.replace(/\r\n/g, '\n');
+        }
+        return text;
+    }
+
+    /**
+     * Indents the given string
+     * @param str  The string to be indented.
+     * @param numOfIndents  The amount of indentations to place at the
+     *     beginning of each line of the string.
+     * @return   The new string with each line beginning with the desired
+     *     amount of indentation.
+     */
+    private indent(str: string, numOfIndents: number): string {
+        // tslint:disable-next-line:prefer-array-literal
+        str = str.replace(/^(?=.)/gm, new Array(numOfIndents + 1).join('\t'));
```

Comment:
Isn't `new Array(numOfIndents + 1).join('\t')` the same as `'\t'.repeat(numOfIndents)`?

Reply:
Fixed.